doc/adr/0004-release-with-continuous-deployment.md

```diff
@@ -0,0 +1,66 @@
+# 4. Release with continuous deployment
+
+Date: 2023-06-22
+
+## Status
+
+TBC
+
+## Context
+
+The Accredited programmes service is currently about to launch a private beta,
+piloting with a small number of prisons, and as such we are focused on
+delivering quickly which requires as little release overhead as possible.
+
+MOJ provide the [tools for continuous
+deployment](https://user-guide.cloud-platform.service.justice.gov.uk/documentation/deploying-an-app/using-circleci-for-continuous-deployment.html),
+configured as part of the
+[TypeScript](https://github.com/ministryofjustice/hmpps-template-typescript/blob/eac50e97924df530e1649dc4ccd4339f4491789d/.circleci/config.yml#L148)
+and
+[Kotlin](https://github.com/ministryofjustice/hmpps-template-kotlin/blob/4c7bc649f8bf8deb8dc49af58c90fd1e9e085437/.circleci/config.yml#L66)
+template repositories. This is an approach taken by other projects at MOJ.
+
+As an external team we need to be mindful that when handing over there will be
+advantages to having a release process consistent with other MOJ services.
+
+We have seen other teams that [have diverged from continuous
+deployment](https://dsdmoj.atlassian.net/wiki/spaces/MRD/pages/4228547717/Release+process)
+and are releasing in batches. We are concerned that adding a heavier process
+like this while we’re still in private beta may introduce overheads that will
+slow down delivery.
+
+## Decision
+
+We will release using continuous deployment. The author of a pull request should
+be responsible for releasing their change through to production before
+considering the work done. We'll perform both manual and automated testing on
```

Comment:
So we want to deploy after every PR rather than through release PRs?

Reply:
On your first question, ideally yes. There may be times that that doesn't apply, but I'm hoping we can make those a rare occurrence.